src/ne2k.js

```diff
@@ -125,7 +125,9 @@ function Ne2k(cpu, bus)
         this.memory[i << 1] = this.memory[i << 1 | 1] = mac[i];
     }
 
-    this.memory[14] = this.memory[15] = 0x57;
+    for(var i = 28; i < 32; i++) {
+      this.memory[i] = 0x57;
+    }
```

Comment:
I'm not an expert in this either, but I think QEMU seems to be setting `0x57` into bytes 14 and 15 as well as bytes 28-31, but here we seem to be setting `0x57` only into bytes 28-31?

Since you were trying to copy over QEMU's behaviour, I thought it might be good to double check whether this deviation from QEMU was intentional.

<sub>(I didn't spend much time reading QEMU's source code, so I could be wrong)</sub>

Reply:
My understanding is that the PROM is 'doubled' for 16-bit cards, which apparently means the same byte is duplicated for a 2-byte word?

So:
```
byte[2 * 14] = byte[2 * 14 + 1] = 0x57
byte[2 * 15] = byte[2 * 15 + 1] = 0x57
```
(Maybe it should be written out like that instead of the loop)

In QEMU, this doubling is a applied for all bytes in reverse so: 
```
byte[14] == byte[7 * 2] 
byte[15] == byte[7 * 2 + 1]
```
The value of byte[14] and byte[15] ends up being byte[7] *after* the doubling.

A quick search on this confirms seems to confirm the doubling (includes an example 32-byte PROM):
https://tldp.org/HOWTO/Ethernet-HOWTO-2.html

Comment in linux kernel: https://github.com/torvalds/linux/blob/master/drivers/net/ethernet/8390/ne2k-pci.c#L199

Reply:
Gotcha. I see, I must have misread QEMU's loop (I didn't realise it overwrites bytes 14 and 15 during the doubling - very sneaky). Thanks, I learnt something new today.